src/clusterfuzz/_internal/tests/core/metrics/logs_test.py

```diff
@@ -463,6 +469,36 @@ def test_error(self):
             }
         })
 
+  def test_log_context(self):
+    """Test log error."""
+    logger = mock.MagicMock()
+    self.mock.get_logger.return_value = logger
+
+    with logs.task_stage_context(logs.Stage.PREPROCESS):
+      statement_line = inspect.currentframe().f_lineno + 1
+      logs.emit(logging.ERROR, 'msg', exc_info='ex', target='bot', test='yes')
+
+    logger.log.assert_called_once_with(
+        logging.ERROR,
+        'msg',
+        exc_info='ex',
+        extra={
+            'extras': {
+                'target': 'bot',
+                'test': 'yes',
+                'task_id': 'f61826c3-ca9a-4b97-9c1e-9e6f4e4f8868',
+                'task_name': 'fuzz',
+                'stage': 'preprocess'
+            },
+            'location': {
+                'path': os.path.abspath(__file__).rstrip('c'),
+                'line': statement_line,
+                'method': 'test_log_context'
+            },
+            'release': 'prod',
+            'docker_image': ''
+        })
+
```

Comment:
Should we assert internal state of the Singleton (such as looking into the metadata dict) or this will make this test brittle? I think that something like `self.assertIn("stage", logs.log_contexts)` for instance and other `self.assert...(..., logs_contexts...)` makes sense.

Reply:
Ops, actually the dictionary key gets deleted after the log, right? Maybe, check if the `meta` dict is empty then? Actually, because it's a singleton, maybe it won't be empty because of other tests?